File: Class 2/Problem 53/maximumSubarray.js
var maxSubArray = function (nums) {
  let currentSum = 0;
  let totalSum = nums[0];

  for (const value of nums) {
    currentSum = Math.max(currentSum + value, value);
    totalSum = Math.max(totalSum, currentSum);
  }

  return totalSum;
};

File: Class 2/Problem 53/maximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        currentSum = 0
        totalSum = nums[0]
        
        for i in range(0, len(nums)):  
           currentSum = max(currentSum + nums[i], nums[i])
           totalSum = max(totalSum, currentSum)
        
        return totalSum
        